include/ginkgo/core/matrix/batch_dense.hpp

```diff
@@ -1010,6 +1010,101 @@ std::unique_ptr<Matrix> batch_initialize(
                                     std::forward<TArgs>(create_args)...);
 }
 
+/**
+ * Creates and initializes a matrix from copies of a given matrix.
+ *
+ * This function first creates a temporary batch dense matrix, fills it with
+ * passed in values, and then converts the matrix to the requested type.
+ *
+ * @tparam Matrix  matrix type to initialize
+ *                 (Dense has to implement the ConvertibleTo<Matrix> interface)
+ * @tparam TArgs  argument types for Matrix::create method
+ *                (not including the implied Executor as the first argument)
+ *
+ * @param stride  row strides for the temporary batch dense matrix
+ * @param num_matrices  The number of times the input matrix is copied into
+ *                     the final output
+ * @param vals  values used to initialize each vector in the temp. batch
+ * @param exec  Executor associated to the vector
+ * @param create_args  additional arguments passed to Matrix::create, not
+ *                     including the Executor, which is passed as the first
+ *                     argument
+ *
+ * @ingroup LinOp
+ * @ingroup mat_formats
+ */
+template <typename Matrix, typename... TArgs>
+std::unique_ptr<Matrix> batch_initialize(
```

Comment:
I am not sure, but maybe you need to combine this with the previous initialize where we have the duplicate the vector ? Because otherwise there might be ambiguity here.